src/bsp/platform/iotc_bsp_io_net_test.cc

```diff
@@ -106,23 +111,19 @@ TEST_P(ServerTest, EndToEndCommunicationWorks) {
             IOTC_BSP_IO_NET_STATE_OK);
   EXPECT_STREQ(recv_buf_, kTestMsg);
 
-  test_server->Stop();
   iotc_bsp_io_net_close_socket(&test_socket_);
+  test_server->Stop();
 }
 
 INSTANTIATE_TEST_CASE_P(
     NetTestSuite, ServerTest,
-    ::testing::Values(new NetworkType{SOCKET_STREAM, PROTOCOL_IPV4,
-                                      const_cast<char*>("127.0.0.1")},
-                      new NetworkType{SOCKET_STREAM, PROTOCOL_IPV6,
-                                      const_cast<char*>("::1")},
-                      new NetworkType{SOCKET_DGRAM, PROTOCOL_IPV4,
-                                      const_cast<char*>("127.0.0.1")},
-                      new NetworkType{
-                          SOCKET_DGRAM,
-                          PROTOCOL_IPV6,
-                          const_cast<char*>("::1"),
-                      }));
+    ::testing::Values(
+        NetworkType(SOCKET_STREAM, PROTOCOL_IPV4,
+                    const_cast<char*>("127.0.0.1")),
```

Comment:
Do you need to pass the address at all? I mean there will be no tests that use an address other than localhost. Based on the protocol the test can figure out the right address, no?

Reply:
Yes, I think I need to pass the address to listen to the right one when if there are  connections from multiple addresses.